src/Nethermind/Nethermind.Core/Collections/SpanConcurrentDictionary.cs

```diff
@@ -327,7 +327,7 @@ private bool TryRemoveInternal(ReadOnlySpan<TKey> key, [MaybeNullWhen(false)] ou
                     }
 
                     Node? prev = null;
-                    for (Node? curr = bucket; curr != null; curr = curr._next)
+                    for (Node? curr = bucket; curr is not null; curr = curr._next)
```

Comment:
one thing about this class - this was copied directly from .net source, not sure if we want to keep it as close as possible as it can work with generic types or are fine with changes here.

Reply:
Are object types not T types or C# would fail to compile on the `is null` test